pkg/framework/pod.go

```diff
@@ -165,6 +167,24 @@ func (cb *ContainerBuilder) Build() v1.Container {
 	return cb.c
 }
 
+func (c *ContextData) GetPodName(label string) (string, error) {
+	podListOpts := metav1.ListOptions{}
+	podListOpts.LabelSelector = "name=" + label
+	podList, err := c.Clients.KubeClient.CoreV1().Pods(c.Namespace).List(podListOpts)
+	if err != nil {
+		return "", err
+	}
+	if podList.Size() == 0 {
+		return "", fmt.Errorf("no pod with label %s found", label)
+	} else if len(podList.Items) > 1 {
+		for _, pod := range podList.Items {
+			log.Logf("Contains %s", pod.Name)
+		}
+		return "", fmt.Errorf("too many pods with label %s found", label)
+	}
+	return podList.Items[0].Name, nil
```

Comment:
Since this is using the label only, I believe it might be usual
to find more than 1 pod with that label. Maybe it would be 
better to return a string slice, instead.